libqtile/layout/plasma.py

```diff
@@ -836,6 +837,18 @@ def __init__(self, **config):
         self._add_mode = None
         Node.priority = Priority.BALANCED if self.fair else Priority.FIXED
 
+    def swap(self, c1: Window, c2: Window) -> None:
+        node_c1 = node_c2 = None
+        for leaf in self.root.all_leafs:
+            if leaf.payload is not None:
+                if c1 == leaf.payload:
+                    node_c1 = leaf
+                elif c2 == leaf.payload:
+                    node_c2 = leaf
+            if node_c1 is not None and node_c2 is not None:
+                node_c1.payload, node_c2.payload = node_c2.payload, node_c1.payload
+                return
```

Comment:
```suggestion
                node_c1.payload, node_c2.payload = node_c2.payload, node_c1.payload
                self.group.layout_all()
                self.group.focus(c1)
                return
```
This should fix it for you. The windows were swapping but you weren't redrawing the layout.

Reply:
That works, thanks. Can this be merged now?